koku/masu/api/aws_bill_cleanup.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2024 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+import logging
+from collections import defaultdict
+from uuid import uuid4
+
+from django.views.decorators.cache import never_cache
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.decorators import permission_classes
+from rest_framework.decorators import renderer_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+from api.common import log_json
+from api.provider.models import Provider
+from masu.processor import is_customer_large
+from masu.processor.tasks import cleanup_aws_bills as cleanup_aws_bills_task
+from masu.processor.tasks import PRIORITY_QUEUE
+from masu.processor.tasks import PRIORITY_QUEUE_XL
+from masu.processor.tasks import QUEUE_LIST
+
+
+LOG = logging.getLogger(__name__)
+REPORT_DATA_KEY = "cleanup_aws_bills Task IDs"
+
+
+@never_cache
+@api_view(http_method_names=["GET"])
+@permission_classes((AllowAny,))
+@renderer_classes(tuple(api_settings.DEFAULT_RENDERER_CLASSES))
+def aws_bill_cleanup(request):
+    """Cleanup AWS bills with a NULL payer account_id."""
+    params = request.query_params
+    schema_name = params.get("schema")
+    bill_date = params.get("bill_date")
+    fallback_queue = PRIORITY_QUEUE
+    if is_customer_large(schema_name):
+        fallback_queue = PRIORITY_QUEUE_XL
+    queue_name = params.get("queue") or fallback_queue
+
+    if queue_name not in QUEUE_LIST:
+        errmsg = f"'queue' must be one of {QUEUE_LIST}."
+        return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
+    if bill_date is None:
+        errmsg = "bill_date is a required parameter."
+        return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Gather all the errors and return them all. This provides a better user experience if more than one parameter is missing.
```suggestion
    errors = []
    if queue_name not in QUEUE_LIST:
        errors.append(f"'queue' must be one of {QUEUE_LIST}.")

    if bill_date is None:
        errors.append("bill_date is a required parameter.")

    if errors:
        return Response([{"Error": message} for message in errors], status=status.HTTP_400_BAD_REQUEST)
```